putty.go

```diff
@@ -53,7 +58,99 @@ type Key struct {
 	Comment           string
 	Encryption        string
 	PrivateMac        []byte
-	decrypted         bool
+	padded            bool
+	keySize           int
+}
+
+func (k Key) Clone() *Key {
+	return &Key{
+		Version:           k.Version,
+		Algo:              k.Algo,
+		PublicKey:         k.PublicKey,
+		PrivateKey:        k.PrivateKey,
+		KeyDerivation:     k.KeyDerivation,
+		Argon2Memory:      k.Argon2Memory,
+		Argon2Passes:      k.Argon2Passes,
+		Argon2Parallelism: k.Argon2Parallelism,
+		Argon2Salt:        k.Argon2Salt,
+		Comment:           k.Comment,
+		Encryption:        k.Encryption,
+		PrivateMac:        k.PrivateMac,
+		//padded:            k.padded,
+		keySize: k.keySize,
+	}
+}
+
+var noLines = strings.NewReplacer("\r", "", "\n", "")
```

Comment:
I think `noNewLines` is a better name for this pattern.

Reply:
Maybe strip line endings as both the carriage return and new lines are stripped

Reply:
After re-reading it, I find I was wrong and you are right.  `noNewLines` is best.  Thank you!